src/helpers/swapErc20.ts

```diff
@@ -0,0 +1,592 @@
+      { internalType: "uint256", name: "nonce", type: "uint256" },
+      { internalType: "uint256", name: "expiry", type: "uint256" },
+      { internalType: "address", name: "signerWallet", type: "address" },
+      { internalType: "address", name: "signerToken", type: "address" },
+      { internalType: "uint256", name: "signerAmount", type: "uint256" },
+      { internalType: "address", name: "senderToken", type: "address" },
+      { internalType: "uint256", name: "senderAmount", type: "uint256" },
+      { internalType: "uint8", name: "v", type: "uint8" },
+      { internalType: "bytes32", name: "r", type: "bytes32" },
+      { internalType: "bytes32", name: "s", type: "bytes32" },
+    ],
+    name: "swap",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      { internalType: "address", name: "recipient", type: "address" },
+      { internalType: "uint256", name: "nonce", type: "uint256" },
+      { internalType: "uint256", name: "expiry", type: "uint256" },
+      { internalType: "address", name: "signerWallet", type: "address" },
+      { internalType: "address", name: "signerToken", type: "address" },
+      { internalType: "uint256", name: "signerAmount", type: "uint256" },
+      { internalType: "address", name: "senderToken", type: "address" },
+      { internalType: "uint256", name: "senderAmount", type: "uint256" },
+      { internalType: "uint8", name: "v", type: "uint8" },
+      { internalType: "bytes32", name: "r", type: "bytes32" },
+      { internalType: "bytes32", name: "s", type: "bytes32" },
+    ],
+    name: "swapAnySender",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      { internalType: "uint256", name: "nonce", type: "uint256" },
+      { internalType: "uint256", name: "expiry", type: "uint256" },
+      { internalType: "address", name: "signerWallet", type: "address" },
+      { internalType: "address", name: "signerToken", type: "address" },
+      { internalType: "uint256", name: "signerAmount", type: "uint256" },
+      { internalType: "address", name: "senderToken", type: "address" },
+      { internalType: "uint256", name: "senderAmount", type: "uint256" },
+      { internalType: "uint8", name: "v", type: "uint8" },
+      { internalType: "bytes32", name: "r", type: "bytes32" },
+      { internalType: "bytes32", name: "s", type: "bytes32" },
+    ],
+    name: "swapLight",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [{ internalType: "address", name: "newOwner", type: "address" }],
+    name: "transferOwnership",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+];
+
+export const getSwapErc20Contract = (
+  providerOrSigner: ethers.providers.Provider | ethers.Signer,
+  chainId: number
+): ethers.Contract => {
+  if (!shouldOverride || chainId !== 1) {
```

Comment:
Let's check whether `chainId` is in `mainnets` from `@airswap/constants` so that all prod chains would be overridden.